airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -63,6 +64,55 @@
   private final ServerSocket stdoutServerSocket;
   private final ExecutorService executorService;
 
+  // TODO(Davin): Cache this result.
+  public static String getCommandFromImage(KubernetesClient client, String imageName) throws IOException, InterruptedException {
+    final String suffix = RandomStringUtils.randomAlphabetic(5).toLowerCase();
+
+    final String podName = "airbyte-command-fetcher-" + suffix;
+
+    Container commandFetcher = new ContainerBuilder()
+        .withName("airbyte-command-fetcher")
+        .withImage(imageName)
+        .withCommand("sh", "-c", "echo \"AIRBYTE_ENTRYPOINT=$AIRBYTE_ENTRYPOINT\"")
+        .build();
+
+    Pod pod = new PodBuilder()
+        .withApiVersion("v1")
+        .withNewMetadata()
+        .withName(podName)
+        .endMetadata()
+        .withNewSpec()
+        .withRestartPolicy("Never")
+        .withContainers(commandFetcher)
+        .endSpec()
+        .build();
+    LOGGER.info("Creating pod...");
+    Pod podDefinition = client.pods().inNamespace("default").createOrReplace(pod);
+    LOGGER.info("Waiting until command fetcher pod completes...");
+    client.resource(podDefinition).waitUntilCondition(p -> p.getStatus().getPhase().equals("Succeeded"), 20, TimeUnit.SECONDS);
+
+    var logs = client.pods().inNamespace("default").withName(podName).getLog();
+    if (!logs.contains("AIRBYTE_ENTRYPOINT")) {
+      // this should not happen
+      throw new RuntimeException("Unable to read AIRBYTE_ENTRYPOINT from the image. Make sure this environment variable is set in the Dockerfile!");
```

Comment:
We should use different logs for each of these errors so we can differentiate when debugging.